src/steps/sse.ts

```diff
@@ -103,12 +106,35 @@ export default async function (
       }
 
       resolve(true)
+    }
+
+    const timeout = setTimeout(() => {
+      console.debug('SSE timed out')
+      end()
     }, params.timeout || 10000)
 
     ev.onerror = (error) => {
       clearTimeout(timeout)
+
+      let message: string
+      if (ev.readyState === EventSource.CLOSED) {
+        // SSE stream closed gracefully
+        return end()
+      } else if (ev.readyState === EventSource.CONNECTING) {
+        // SSE stream closed by the server
+        if (expectedMessages === undefined) {
+          message = 'The SSE stream was closed by the server. If this is expected behavior, please use [`tests.<test>.steps.[step].sse.check.messages`](https://docs.stepci.com/reference/workflow-syntax.html#tests-test-steps-step-sse-check-messages-message).'
+        } else {
+          message = `The SSE stream was closed by the server before all expected messages were received. Missing IDs: ${JSON.stringify([...expectedMessages], null, 2)}`
+        }
+      } else {
+        // SSE stream is still open (`ev.readyState === EventSource.OPEN`)
+        // but received an "error" event from the server
+        message = `The SSE stream received an error event from the server: ${JSON.stringify(error, null, 2)}`
+      }
+
       ev.close()
-      reject(error)
+      reject({ ...error, message })
```

Comment:
Does it look good in the CLI?

Reply:
Yes it does, it integrates with `stepci` reading the `message` key. That's why I did it in the first place (it printed an unhelpful "undefined" before, as I stated in the first paragraph of this PR description).

Reply:
(I let you resolve conversations)